tests/testthat/test-util.R

```diff
@@ -698,3 +698,93 @@ test_that("random_seed", {
   expect_equal(random_seed(e1), pi)
   expect_null(random_seed(e2))
 })
+
+
+## TODO: add tests of some of the comment processing
+test_that("yaml_block_info - simple test", {
+  yml <- c("a:", "  - 1", "  - 2", "b:", "    3", "c: 4")
+  expect_equal(yaml_load(yml), list(a = 1:2, b = 3, c = 4))
+
+  expect_equal(yaml_block_info("a", yml),
+               list(name = "a", exists = TRUE, block = TRUE,
+                    start = 1L, end = 3L, indent = "  "))
+  expect_equal(yaml_block_info("b", yml),
+               list(name = "b", exists = TRUE, block = TRUE,
+                    start = 4L, end = 5L, indent = "    "))
+  expect_equal(yaml_block_info("c", yml),
+               list(name = "c", exists = TRUE, block = FALSE,
+                    start = 6L, end = 6L))
+  expect_equal(yaml_block_info("d", yml),
+               list(name = "d", exists = FALSE, block = FALSE))
+})
+
+
+test_that("yaml parse failure", {
+  text <- c("a: 1", "a: 2")
+  expect_error(yaml_load(text))
+  expect_error(yaml_block_info("a", text), "Failed to process yaml")
+})
+
+
+test_that("insert into files", {
+  text <- c("a", "b", "c")
+  value <- c("x", "y")
+  path <- tempfile()
+  writeLines(text, path)
+
+  expect_message(
+    str <- capture.output(
```

Comment:
This is a new thing for me and shows up something I haven't seen before. When I've done things like this where I want to write tests about the messages printed and the result returned I've used `testthat::evaluate_promise` which I think is doing something similar to here but it returns the `result`, `message` and the `output`. So I wasn't aware the was a difference here.

What is the difference between `output` and `result`? Result is the return from the function and the `output` is what is coming from call to `cat`. Perhaps I don't quite get how `cat` works. How is `cat` different from a `message` or from `print`?

Reply:
I was looking for `testthat::evaluate_promise` but could not find it - I'll rewrite to use that as it will be clearer.

cat writes to stdout, message to stderr, print *probably* writes to stdout, but it relies on underlying cat/message calls to do the printing